.github/workflows/build-yocto.yml

```diff
@@ -1,12 +1,13 @@
-name: Push Build
+name: Build Yocto
 
 on:
+  pull_request:
```

Comment:
We had push as a separated file as we also want to implement https://github.com/quic-yocto/meta-qcom-hwe/issues/21, which needs a post build action of saving the yml lock file (similar to pinned manifest), and I believe we might want to save just for push (we could save for prs as well, but not sure if that is valuable).

We can combine now and split later if a new job turns out to be needed.

Reply:
we can drop this PR if you want. Its not a big deal.